src/core/Akka/Actor/ActorRef.cs

```diff
@@ -131,7 +131,15 @@ protected override void TellInternal(object message, IActorRef sender)
             {
                 if (Interlocked.Exchange(ref status, COMPLETED) == INITIATED)
                 {
-                    _result.TrySetResult(message);
+                    if (message is T t)
+                    {
+                        _result.TrySetResult(t);
+                    }
+                    else
+                    {
+                        _result.TrySetException(new ArgumentException(
+                            $"Received message of type [{message.GetType()}] - Ask expected message of type [{typeof(T)}]"));
```

Comment:
In the past this error was raised as an `InvalidCastException` - changed to `ArgumentException`